src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminal.accessibility.contribution.ts

```diff
@@ -32,18 +32,36 @@ class TextAreaSyncContribution extends DisposableStore implements ITerminalContr
 	static get(instance: ITerminalInstance): TextAreaSyncContribution | null {
 		return instance.getContribution<TextAreaSyncContribution>(TextAreaSyncContribution.ID);
 	}
+	private _addon: TextAreaSyncAddon | undefined;
 	constructor(
 		private readonly _instance: ITerminalInstance,
 		processManager: ITerminalProcessManager,
 		widgetManager: TerminalWidgetManager,
-		@IInstantiationService private readonly _instantiationService: IInstantiationService
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService
 	) {
 		super();
 	}
+
 	xtermReady(xterm: IXtermTerminal & { raw: Terminal }): void {
-		const addon = this._instantiationService.createInstance(TextAreaSyncAddon, this._instance.capabilities);
-		xterm.raw.loadAddon(addon);
-		addon.activate(xterm.raw);
+		if (this._configurationService.getValue(TerminalSettingId.SyncTextArea)) {
```

Comment:
Do we enable this one when screen reader mode is on? If not, we should be doing that

Reply:
We only hook up the listeners when that's on but yeah maybe best to not even create it until that happens.

Reply:
It's fine to create the addon as it's very lightweight if it doesn't do anything. I was expecting to see the accessibility service in the addon though, not sure where that check is done

Reply:
here's where we handle that

https://github.com/microsoft/vscode/blob/d266a16cfa78ef3b3c0005bbaf3c80f7d05ebec1/src/vs/workbench/contrib/terminalContrib/accessibility/browser/textAreaSyncAddon.ts#L25-L46